lib/quicly.c

```diff
@@ -2532,15 +2541,15 @@ int64_t quicly_get_first_timeout(quicly_conn_t *conn)
             return 0;
         if (scheduler_can_send(conn))
             return 0;
-    } else if (!conn->super.remote.address_validation.validated) {
-        return conn->idle_timeout.at;
     }
 
-    int64_t at = conn->egress.loss.alarm_at;
-    if (conn->egress.send_ack_at < at)
-        at = conn->egress.send_ack_at;
-    if (conn->idle_timeout.at < at)
-        at = conn->idle_timeout.at;
+    int64_t at = conn->idle_timeout.at;
```

Comment:
Can you add a comment above saying something like "Set the PTO timer if data can be sent, else set idle timer."

Reply:
Done in f8e46c2.